llvm-project/libcxx/src/exception.cpp:17:12: fatal error: 'cxxabi.h' file not found

Same issue as #82197. 

<details>

<summary>Full error</summary>

```log
FAILED: libcxx/src/CMakeFiles/cxx_shared.dir/exception.cpp.o
/Users/kip/llvm-project/build/bin/clang++ --target=arm64-apple-darwin23.5.0 -DLIBCXX_BUILDING_LIBCXXABI -DLIBC_NAMESPACE=__llvm_libc_common_utils -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_LIBCPP_BUILDING_LIBRARY -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBCPP_REMOVE_TRANSITIVE_INCLUDES -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/Users/kip/llvm-project/libcxx/src -I/Users/kip/llvm-project/build/include/c++/v1 -I/Users/kip/llvm-project/cmake/Modules/../../libc -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -g -std=c++23 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.2.sdk -mmacosx-version-min=14.5 -fPIC -faligned-allocation -nostdinc++ -fvisibility-inlines-hidden -fvisibility=hidden -fsized-deallocation -Wall -Wextra -Wnewline-eof -Wshadow -Wwrite-strings -Wno-unused-parameter -Wno-long-long -Werror=return-type -Wextra-semi -Wundef -Wunused-template -Wformat-nonliteral -Wzero-length-array -Wdeprecated-redundant-constexpr-static-def -Wno-nullability-completeness -Wno-user-defined-literals -Wno-covered-switch-default -Wno-suggest-override -Wno-error -MD -MT libcxx/src/CMakeFiles/cxx_shared.dir/exception.cpp.o -MF libcxx/src/CMakeFiles/cxx_shared.dir/exception.cpp.o.d -o libcxx/src/CMakeFiles/cxx_shared.dir/exception.cpp.o -c /Users/kip/llvm-project/libcxx/src/exception.cpp
/Users/kip/llvm-project/libcxx/src/exception.cpp:17:12: fatal error: 'cxxabi.h' file not found
   17 | #  include <cxxabi.h>
      |            ^~~~~~~~~~
1 error generated.
```

</details>

Commands:

```sh
$ mkdir build && cd build

$ cmake -DCMAKE_INSTALL_PREFIX=$(pwd)/install-prefix -DCMAKE_C_COMPILER=/opt/homebrew/Cellar/llvm/20.1.2/bin/clang -DCMAKE_CXX_COMPILER=/opt/homebrew/Cellar/llvm/20.1.2/bin/clang++ -DCMAKE_BUILD_TYPE=Debug -DLLVM_ENABLE_PROJECTS="clang;lldb"  -DLLVM_TARGETS_TO_BUILD=AArch64 -DLLVM_USE_LINKER=lld -DCMAKE_CXX_STANDARD=20 -DLLVM_BUILD_TESTS=ON -DLLVM_ENABLE_RUNTIMES=libcxx -GNinja ../llvm

$ ninja
```

Configuration:
- Mac M1
- MacOS Sonoma 14.5
- Darwin Kernel Version 23.5.0
- Git commit e8245d53247f155d234e1de647aac2678c49ac28

The same problem on x86 (llvm-20.1.4):
```bash
CFLAGS="-O3 -march=x86-64-v3 --gcc-toolchain=/opt/gcc-15.1.0"                   \
CXXFLAGS="-O3 -march=x86-64-v3 -stdlib=libc++ --gcc-toolchain=/opt/gcc-15.1.0"  \
cmake                                                \
        -DCMAKE_BUILD_TYPE=Release                   \
        -DCMAKE_INSTALL_PREFIX=/opt/llvm-20.1.4      \
        -DLLVM_ENABLE_PROJECTS="clang;lld;clang-tools-extra;polly" \
        -DLLVM_ENABLE_RUNTIMES="libcxx;compiler-rt"  \
        -DCMAKE_CXX_STANDARD=20                      \
        -DCMAKE_C_COMPILER=clang                     \
        -DCMAKE_CXX_COMPILER=clang++                 \
        -DCMAKE_EXE_LINKER_FLAGS="-stdlib=libc++ -rtlib=compiler-rt -lc++ -lc++abi -lunwind"     \
        -DCMAKE_SHARED_LINKER_FLAGS="-stdlib=libc++ -rtlib=compiler-rt -lc++ -lc++abi -lunwind"  \
        -DCMAKE_INSTALL_RPATH='$ORIGIN/../lib/x86_64-unknown-linux-gnu:$ORIGIN/../lib'           \
        -DLLVM_ENABLE_LIBCXX=ON                      \
        -DLLVM_ENABLE_LLD=ON                         \
        -DLLVM_ENABLE_LTO=OFF                        \
        -DLLVM_TARGETS_TO_BUILD=X86                  \
        -DLLVM_ENABLE_RTTI=ON                        \
        -DLLVM_ENABLE_ZSTD=FORCE_ON                  \
        -DLLVM_USE_STATIC_ZSTD=ON                    \
        -DLLVM_ENABLE_ZLIB=FORCE_ON                  \
        -DLIBCXX_USE_COMPILER_RT=YES                 \
        -DLIBCXXABI_USE_COMPILER_RT=YES              \
        -DLIBCXX_HERMETIC_STATIC_LIBRARY=ON          \
        -DLIBCXX_ENABLE_STATIC_ABI_LIBRARY=ON        \
        -DLIBCXXABI_ENABLE_STATIC_UNWINDER=ON        \
        -DZSTD_ROOT="/opt/zstd-1.5.7"                \
        -DZLIB_ROOT="/opt/zlib-1.3.1"                \
        -S ./llvm-20.1.4/llvm                        \
        -B ./llvm-buildroot
```
...and the error:
```
/root/llvm-20.1.4/libcxx/src/exception.cpp:17:12: fatal error: 'cxxabi.h' file not found
   17 | #  include <cxxabi.h>
      |            ^~~~~~~~~~
```

--- a/libcxx/cmake/Modules/HandleLibCXXABI.cmake
+++ b/libcxx/cmake/Modules/HandleLibCXXABI.cmake
@@ -148,6 +148,12 @@ elseif ("${LIBCXX_CXX_ABI}" STREQUAL "libcxxabi")
     add_library(libcxx-abi-static-objects ALIAS cxxabi_static_objects)
   endif()
 
+  if (NOT TARGET cxxabi_static AND NOT TARGET cxxabi_shared
+      AND NOT TARGET cxxabi_static_objects AND NOT TARGET cxxabi_shared_objects)
+    message(FATAL_ERROR "Can't find any libc++abi target but LIBCXX_CXX_ABI is set to libcxxabi, "
+                        "did you forget to include libcxxabi in LLVM_ENABLE_RUNTIMES?")
+  endif()
+
 # Link against a system-provided libc++abi
 elseif ("${LIBCXX_CXX_ABI}" STREQUAL "system-libcxxabi")
   if(NOT LIBCXX_CXX_ABI_INCLUDE_PATHS)

--- a/libcxx/cmake/Modules/HandleLibCXXABI.cmake
+++ b/libcxx/cmake/Modules/HandleLibCXXABI.cmake
@@ -151,7 +151,8 @@ elseif ("${LIBCXX_CXX_ABI}" STREQUAL "libcxxabi")
   if (NOT TARGET cxxabi_static AND NOT TARGET cxxabi_shared
       AND NOT TARGET cxxabi_static_objects AND NOT TARGET cxxabi_shared_objects)
     message(FATAL_ERROR "Can't find any libc++abi target but LIBCXX_CXX_ABI is set to libcxxabi, "
-                        "did you forget to include libcxxabi in LLVM_ENABLE_RUNTIMES?")
+                        "did you forget to include libcxxabi in LLVM_ENABLE_RUNTIMES, or did you "
+                        "intend to use LIBCXX_CXX_ABI=system-libcxxabi instead?")
   endif()
 
 # Link against a system-provided libc++abi

